File: CPP/Solution961.cpp
using namespace std;
class Solution {
public:
    int repeatedNTimes(vector<int> &A) {
        for (int i = A.size() - 1; i > 1; --i) {
            if (A[i] == A[i - 1] or A[i] == A[i - 2]) {
                return A[i];
            }
        }
        return A[0];
    }
};

File: src/Solution961.java
import java.util.HashMap;
import java.util.Map;

/**
 * Created by slade on 2019/11/30.
 */
class Solution961 {
    public int repeatedNTimes(int[] A) {
        int length = A.length >> 1;
        Map hashmap = new HashMap<>();
        for (int item : A) {
            if (hashmap.containsKey(item)) {
                int value = (int) hashmap.get(item);
                if (value + 1 == length) {
                    return item;
                }
                hashmap.put(item, value + 1);
            } else {
                hashmap.put(item, 1);
            }
        }
        return -1;
    }
}
